### DISCUSSION TITLE
Defining metric multiple times

### DISCUSSION THREAD

**Discussion by NeoCortex97** [prometheus/client_python#850] [General]
Hi,
I wanted to ask, if it is safe to define the same metrics multiple times, or should i make a separate module for global metrics and import them everywhere i want them to be used?

Or should I rather have specialized metrics for every place i want it to be used and merge them in queries?

**Comment by csmarchbanks** [comment]
Typically you will not want to define the same metric multiple times, and should probably be using different metrics in different parts of your code base. As you say, these can then be merged in queries.

If you do need a shared metric then a separate module is probably the best way to go. If you have a concrete example I am happy to give you guidance on it!

**Reply by NeoCortex97** [reply]
My current Problem is, that I am building a discord bot that should track the number of commands executed total, but it has multiple things called cogs, that are loaded dynamically. 

I could simply overwrite things inside the bot, but trying that lead to strange bugs, so I wanted to solve my problem without doing that.But maybe I really should have separate Metrics for everything.

**Reply by csmarchbanks** [reply]
`commands_executed_total` certainly sounds like it should be a single metric with a `cogs` label. I would probably define the metric in one shared module and use it wherever necessary for this use case. Multiple things loaded dynamically makes following the general rule of thumb a bit harder.
